src/mcpbr/storage/cloud.py

```
@@ -89,17 +210,22 @@ def download(self, key: str, local_path: Path) -> Path:
-                capture_output=True,
-                text=True,
-                check=True,
                 timeout=300,
+                operation_label=f"S3 download from {s3_uri}",
             )
             logger.info(f"Downloaded {s3_uri} to {local_path}")
             return local_path
         except subprocess.CalledProcessError as e:
+            # Clean up any partial/stale file to prevent corrupt data
+            Path(local_path).unlink(missing_ok=True)
             raise CloudStorageError(f"S3 download failed: {e.stderr}") from e
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)